magda-web-client/src/Components/Dataset/Add/Pages/People/CustodianDropdown.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import { useAsync } from "react-async-hook";
+import Select from "react-select";
+import { config } from "config";
+import find from "lodash/find";
+
+import ReactSelectStyles from "Components/Common/react-select/ReactSelectStyles";
+
+import {
+    listOrgUnitsAtLevel,
+    OrgUnitWithRelationship
+} from "api-clients/OrgUnitApis";
+
+type Props = {
+    orgUnitId?: string;
+    teamOrgUnitId?: string;
+    onChange: (orgUnitId: string) => void;
+};
+
+export default function CustodianDropdown({
+    orgUnitId,
+    teamOrgUnitId,
+    onChange: onChangeCallback
+}: Props) {
+    // --- use hasUserSelected to tell whether user has selected a value or not
+    const [hasUserSelected, setHasUserSelected] = useState(false);
+    const { loading, error, result, execute } = useAsync(listOrgUnitsAtLevel, [
+        config.custodianOrgLevel,
+        teamOrgUnitId
+    ]);
+
+    if (loading) {
+        return <span>Loading...</span>;
+    } else if (error || !result || result.length === 0) {
+        return (
+            <div className="au-body au-page-alerts au-page-alerts--error">
+                <span style={{ verticalAlign: "-2px" }}>
+                    Could not retrieve data custodian list, or there are no data
+                    custodians in the system.
+                </span>
+                <button
+                    className="au-btn au-btn--tertiary"
+                    onClick={() =>
+                        execute(config.custodianOrgLevel, teamOrgUnitId)
+                    }
+                >
+                    Try Again
+                </button>
+            </div>
+        );
+    } else {
+        const selectedValue = orgUnitId && find(result, option => option.id === orgUnitId);
+
+        if ((!selectedValue || !hasUserSelected) && teamOrgUnitId) {
+            const relatedOrgUnit = find(
```

Comment:
You could do the cast to `OrgUnitWithRelationship` here to avoid doing it twice later :)

Reply:
Thanks 👍Will have a try~
Previously try to cast it to a (relatedOrgUnit | undefined) but seems typescript not smart enough to let me use `OrgUnitWithRelationship.id`